Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -1407,12 +1487,23 @@ private void ProcessTimeout(double time, int mapIndex, int controlIndex, int bin
             };
 
             currentState.isTimerRunning = false;
-            interactionStates[interactionIndex] = currentState;
+            currentState.totalTimeoutCompletionTimeRemaining =
+                Mathf.Max(currentState.totalTimeoutCompletionTimeRemaining - currentState.timerDuration, 0);
+            currentState.timerDuration = default;
 
             // Let interaction handle timer expiration.
             interactions[interactionIndex].Process(ref context);
         }
 
+        internal void SetTotalTimeoutCompletionTime(float seconds, ref TriggerState trigger)
+        {
+            Debug.Assert(trigger.interactionIndex >= 0 && trigger.interactionIndex < totalInteractionCount, "Interaction index out of range");
```

Comment:
Same thing as above. We should use Assert instead of Debug.Assert to avoid the double error.